client/ayon_applications/plugins/launcher_actions/debug_shell.py

```diff
@@ -0,0 +1,116 @@
+from typing import Optional
+import os
+import subprocess
+
+from qtpy import QtWidgets, QtGui, QtCore
+
+from ayon_applications import (
+    Application,
+    ApplicationManager,
+)
+from ayon_applications.utils import get_app_environments_for_context
+from ayon_core.pipeline import LauncherAction
+from ayon_core.style import load_stylesheet
+
+
+def get_application_qt_icon(
+        application: Application
+) -> Optional[QtGui.QIcon]:
+    """Return QtGui.QIcon for an Application"""
+    # TODO: Improve workflow to get the icon, remove 'color' hack
+    from ayon_core.tools.launcher.models.actions import get_action_icon
+    from ayon_core.tools.utils.lib import get_qt_icon
+    application.color = "white"
+    return get_qt_icon(get_action_icon(application))
```

Comment:
```suggestion
    icon = action.icon
    if not icon:
        return QtGui.QIcon()
    icon_filename = os.path.basename(icon)
    icon_filepath = os.path.join(APPLICATIONS_ADDON_ROOT, "icon", icon_filename)
    if os.path.exists(icon_filepath):
        return QtGui.QIcon(icon_filepath)
    return QtGui.QIcon()
```

Reply:
Doesn't this make it:
- fail to use the resource / icons cache?
- fail to allow awesome-font icons or alike (and its coloring) if anything _would/could_ define that?

Reply:
Hey,
I debugged this for a while, I've an alternative solution.

```suggestion
    from ayon_core.tools.launcher.models.actions import get_action_icon
    from ayon_core.tools.utils.lib import get_qt_icon
    application.color = "white"

    icon = get_qt_icon(get_action_icon(application))
    sizes = icon.availableSizes()
    if sizes:
        if not (sizes[0].toTuple() > (1, 1) ):
            icon_filename = os.path.basename(application.icon)
            icon_filepath = os.path.join(
                APPLICATIONS_ADDON_ROOT, "icons", icon_filename)
            if os.path.exists(icon_filepath):
                return QtGui.QIcon(icon_filepath)
            return QtGui.QIcon()
    return icon
```
![image](https://github.com/user-attachments/assets/56e8f213-fdc9-471c-a5db-dbda5210216f)

Reply:
or maybe 
```suggestion
    from ayon_core.tools.launcher.models.actions import get_action_icon
    from ayon_core.tools.utils.lib import get_qt_icon
    application.color = "white"
    icon = get_qt_icon(get_action_icon(application))
    sizes = icon.availableSizes()
    if not sizes:
        return QtGui.QIcon()
        
    if sizes[0].toTuple() > (1, 1):
        return icon
        
    icon_filename = os.path.basename(application.icon)
    icon_filepath = os.path.join(
        APPLICATIONS_ADDON_ROOT, "icons", icon_filename)
    if os.path.exists(icon_filepath):
        return QtGui.QIcon(icon_filepath)
    
    return QtGui.QIcon()
```

Reply:
Had a look myself - it basically failed to resolve the filepath to the icon correctly because it looked in `ayon_core` resources for the icon, but it's now in ayon applications and I think that's why @iLLiCiTiT provided me the suggestion.

I've now pushed a fix.